src/main/java/hw15/ProductManager.java

```diff
@@ -0,0 +1,57 @@
+package hw15;
+import java.time.LocalDate;
+import java.util.Comparator;
+import java.util.List;
+import java.util.Map;
+import java.util.NoSuchElementException;
+import java.util.stream.Collectors;
+
+public class ProductManager {
+    public static List<Product> getFilteredProducts(List<Product> products) {
+        return products.stream()
+                .filter(product -> ProductCategory.BOOK.equals(product.getCategory()))
+                .filter(product -> product.getPrice() > 250)
+                .toList();
+    }
+
+    public static List<Product> getDiscountedBooks(List<Product> products) {
+        return products.stream()
+                .filter(product -> ProductCategory.BOOK.equals(product.getCategory()))
+                .filter(Product::isDiscountAvailable)
+                .map(product -> new Product(
+                        product.getCategory(),
+                        product.getPrice() * 0.9,
+                        product.isDiscountAvailable(),
+                        product.getDateAdded())
+                )
+                .collect(Collectors.toList());
+    }
+
+    public static Product getCheapestBook(List<Product> products) {
+        return products.stream()
+                .filter(product -> ProductCategory.BOOK.equals(product.getCategory()))
+                .min(Comparator.comparing(Product::getPrice) )
+                .orElseThrow(() -> new NoSuchElementException("No product with category 'Book' found."));
+    }
+
+    public static List<Product> getLastThreeAddedProducts(List<Product> products) {
+        return products.stream()
+                .sorted(Comparator.comparing(Product::getDateAdded).reversed())
+                .limit(3)
+                .collect(Collectors.toList());
```

Comment:
```suggestion
                .toList();
```
те саме, просто менше коду